File: backend/shared/uidl/src/main/java/io/mateu/uidl/data/ContextMenu.java
package io.mateu.uidl.data;

import io.mateu.uidl.fluent.Component;
import io.mateu.uidl.interfaces.Actionable;
import java.util.List;
import lombok.Builder;

@Builder
public record ContextMenu(
    List<Actionable> menu,
    Component wrapped,
    boolean activateOnLeftClick,
    String style,
    String cssClasses)
    implements Component {}

File: frontend/web/vaadin/src/mateu/shared/apiClients/dtos/componentmetadata/ContextMenu.ts
import ComponentMetadata from "@mateu/shared/apiClients/dtos/ComponentMetadata";
import MenuOption from "@mateu/shared/apiClients/dtos/componentmetadata/MenuOption";
import Component from "@mateu/shared/apiClients/dtos/Component";

export default interface ContextMenu extends ComponentMetadata {

    menu: MenuOption[]
    wrapped: Component
    activateOnLeftClick: boolean

}